api/src/main/proto/bio/terra/tanagra/utils/pagination.proto

```diff
@@ -0,0 +1,17 @@
+// Protobuf definitions for service pagination.
+// We base64 encode proto binary representations for next_page_tokens in the
+// API following https://cloud.google.com/apis/design/design_patterns#list_pagination
+
+syntax = "proto3";
+
+package bio.terra.tanagra.utils;
+option java_package = "bio.terra.tanagra.proto.utils.pagination";
+option java_multiple_files = true;
+
+// A simple pagination message with the page size and offset for the page.
+message PageSizeOffset {
+  // The number of results to return per page.
+  int32 page_size = 1;
```

Comment:
This is a good starting point and works well for rarely changing (effectively static) resources like underlay / entitity. For mutable endpoints (for Tanagra, mostly this pertains to artifacts), likely you'll want something more stable, e.g. `next_id`, or for performance, possibly something more tailored the endpoint that allows you to take advantage of the index. Alternatively, having some other way of dealing with insert/deletes of resources mid-pagination stream